runner.py

```diff
@@ -1,138 +1,228 @@
+        "Default value depends on the target",
+    )
+
+    parser.add_argument(
+        "-b",
+        "--baudrate",
+        default=115200,
+        help="Specify the connection speed of serial. " "By default uses %(default)d",
+    )
+
+    parser.add_argument(
+        "--no-flash",
+        default=False,
+        action="store_true",
+        help="Board will not be flashed by runner.",
+    )
+
+    parser.add_argument(
+        "--no-test",
+        default=False,
+        action="store_true",
+        help="Tests will be not run by runner. This option may be helpful to only flash the device "
+        "or check if yamls are parsed correctly."
+    )
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Increases verbosity level",
+    )
+
+    # Add alternative option "--nightly" "" for nightly argument, which is False
     # Needed for proper disabling long tests in gh actions
     # because of problem with passing empty argument through gh workflows
-    parser.add_argument("--long-test",
-                        type=bool,
-                        nargs='?', default=False, const=True,
-                        help="Long tests will be run")
+    parser.add_argument(
+        "--nightly",
+        type=bool,
+        nargs="?",
+        default=False,
+        const=True,
+        help="Nightly tests will be run",
+    )
 
     args = parser.parse_args()
 
-    args.log_level = logging_level[args.log_level]
-
     if not args.test:
-        args.test = [config.PHRTOS_TEST_DIR]
+        args.test = [resolve_project_path()]
 
-    if not args.serial:
-        if args.target == 'armv7a9-zynq7000-zedboard':
-            args.serial = config.DEVICE_SERIAL_PORT_XYLINX
-        else:
-            args.serial = config.DEVICE_SERIAL_PORT_NXP
+    return args
 
-    if not args.baudrate:
-        args.baudrate = config.DEVICE_SERIAL_BAUDRATE
 
-    return args
+def resolve_project_path():
+    file_path = Path(__file__).resolve()
+    # file_path is phoenix-rtos-project/phoenix-rtos-tests/runner.py
+    project_dir = file_path.parent.parent
+    return project_dir
+
+
+def resolve_targets_and_hosts() -> Tuple[Dict[str, TargetBase], Dict[str, Host]]:
+    """
+    Returns a tuple of dictionaries that map host and target names to an equivalent class.
+
+    Load external targets and hosts defined by user in extensions and combine them
+    with internal runnner targets.
+    """
+
+    targets = [
+        HostPCGenericTarget,
+        IA32GenericQemuTarget,
+        RISCV64GenericQemuTarget,
+        ARMv7A9Zynq7000QemuTarget,
+        STM32L4x6Target,
+        IMXRT106xEvkTarget,
+        IMXRT117xEvkTarget,
+        Zynq7000ZedboardTarget,
+    ]
+
+    hosts = [EmulatorHost, RpiHost]
+
+    try:
+        ext_targets, ext_hosts = load_extensions()
+    except ExtensionError as e:
+        print(e)
+        sys.exit(4)
```

Comment:
I do not like concept of extensions. Imo better way is to create designated directories for hosts and targets. load all files from those directories using this trick: https://stackoverflow.com/a/1057534
And then getting all possible targets and host via recusrsive __subclasses__ https://docs.python.org/3/library/stdtypes.html?highlight=__subclasses__#class.__subclasses__

Reply:
After call I undestood the nned for separate folders however, I still think that magic with __subclasses__ would be better